### ISSUE TITLE
Support multidimensional meshes

### ISSUE THREAD

**Issue by rc** [sfepy/sfepy#128]
Currently, single dimensional meshes are supported, so elements of other
dimensions included in the mesh can cause problems. Supporting meshes of
mixed dimensionality may allow features such as defining boundary
conditions on faces or contact elements.<p>Migrated from http://code.google.com/p/sfepy/issues/detail?id=126</p>

**Comment by gustxsr** [comment]
Hello!
I am getting the following NotImplementedError when I try to define the domain of my mesh:
`NotImplementedError: meshes with several cell kinds are not supported!`
Does this have anything to do with the support for multidimensional meshes? My mesh is composed of 2D quadrilateral elements covering the surface area of several hollow sphere-like objects.
Thank you beforehand!
-Gustavo Santiago-Reyes

**Comment by rc** [comment]
So you have 2D elements in 3D space? Currently, such elements need to be used only with dedicated terms (dw_shell10x).

But the error is caused probably by having other types of elements besides the quadrilaterals in your mesh - are there line or triangle elements as well?

**Comment by gustxsr** [comment]
Yes! I included a picture of the mesh.
![image](https://user-images.githubusercontent.com/46436777/61706188-7330b080-ad15-11e9-8e5a-bf7bbd0b646f.png)
I deleted the lines in the Gmsh script and the error was not present anymore! Thank you. However, these warnings appear:
`sfepy: warning: mesh with topological dimension 2 lower than space dimension 3`
`sfepy: - element orientation not checked!`
I will update if anything else comes up, but I am back on track.
Thank you again for your response :)!
